src/OpenDiffix.Core/Aggregator.fs

```diff
@@ -382,6 +382,125 @@ type private DiffixSumNoise(summandType, aidsCount) =
       | Anonymizer.AnonymizedResult.NotEnoughAIDVs -> Null
       | Anonymizer.AnonymizedResult.Ok { NoiseSD = noiseSD } -> Real noiseSD
 
+
+type private CountHistogram(binSize: int64 option) =
+  let state = Dictionary<Value, int64>()
+
+  let generalize count =
+    match binSize with
+    | None
+    | Some 1L -> count
+    | Some binSize -> (float count / float binSize) |> floor |> int64 |> (*) binSize
+
+  interface IAggregator with
+    member this.Transition args =
+      match args with
+      | Null :: _ -> () // Ignore NULLs
+      | aid :: _ -> state |> Dictionary.increment aid
+      | _ -> invalidArgs args
```

Comment:
Do I understand correctly, that multi-AID isn't supported or is it working transparently as a list of AID instances? If the former, we could put in a specific error case for this, if the latter maybe we should spell this `aids` or `aidList` here, to make it clear we support multi-AID.

(same comment for the Diffix version, with the additional twist, that we need to differentiate from the `aidInstances` there. So perhaps, a specific name to put here and there would be `countedAid` / `countedAids`?)

Reply:
> Do I understand correctly, that multi-AID isn't supported or is it working transparently as a list of AID instances?

We pass the list of AIDs like usual, but only _one_ AID can be counted. The rest are used for LCF during pass 2 of processing. As a reminder:

- Pass 1: Gather a `map<aid, (rows_contributed, sibling_aid_sets[])>` during agg transitions where `sibling_aids` are AID sets from rows associated with given `aid`.
- Pass 2: Invert the map to `map<num_rows, (num_aids_with_such_rows, aid_instances[])>`, filter bins with low count `aid_instances`.
- Pass 3: Finalize the map to a sorted list of 2-tuples `(count, num_aids)`.

> So perhaps, a specific name to put here and there would be countedAid / countedAids?

Good idea!

Reply:
gotcha, so it's going to be `countedAid` (singular) or any better alternative you might find, correct?

Reply:
I mean, we can't abuse this by passing a _single_ AID which has a type `List`?

Reply:
> I mean, we can't abuse this by passing a _single_ AID which has a type `List`?

Right now the anon version accepts the following args: `(aidInstances: List, countedAid: Aid, binSize?: integer)`.

`countedAid` is guaranteed to be a member of `aidInstances` but it's simpler to just pass it again.